client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts

```diff
@@ -156,25 +152,3 @@ export function getArrayOfFilteredTasks( tasks: Task[], flow: string | null ) {
 		}, [] as Task[] )
 	);
 }
-
-// This function will determine whether we want to disable or enable a task on the checklist
-// If a task depends on the completion of other tasks, we want to check if all dependencies are finished:
-//    ^ If all the dependencies are done ( true ), then the task is enabled
-//    ^ If at least one of the dependencies is unfinished ( false ), then we check the proceeding conditions
-// If a task is set to keepActive, we keep it enabled. It allows a task to be revisited when completed
-// If a task is completed, we disable it
-export function isTaskDisabled( task: Task ) {
-	if ( hasIncompleteDependencies( task ) ) {
-		return true;
-	}
-
-	if ( task.keepActive ) {
-		return false;
-	}
-
-	return task.isCompleted;
-}
-
-export function hasIncompleteDependencies( task: Task ) {
```

Comment:
the `hasIncompleteDependencies` is no longer necessary since we can modify the `disabled` prop directly, by checking if the task's dependencies are resolved (completed) and setting it to `false`, otherwise setting it to `true`